Archives/VolFile.cpp

```diff
@@ -115,11 +113,11 @@ namespace Archives
 		try
 		{
 			char* offset = (char*)m_IndexEntries[fileIndex].dataBlockOffset;
-			int length = *(int*)(offset + 4) & 0x7FFFFFFF;
+			std::size_t length = *(int*)(offset + 4) & 0x7FFFFFFF;
```

Comment:
File access should use fixed sized fields.

The original code was de-referencing a pointer into the mapped view of the file. Hence it used to be `base + offset`, where base was the base address of the window into the memory mapped file, and offset was the offset into the file.

This data needs to be actually read into memory now.